src/classes/AuthenticationSession.ts

```diff
@@ -32,6 +42,59 @@ export class AuthenticationSession {
     this.promptForPassword();
   }
 
+  public obtainNewAuthTokenUsingRefreshToken(): void {
+    if (!AuthenticationSession.sftpFusionAuthClientId) {
+      logger.error(
+        'Cannot obtain new access token without sftp client id.',
+      );
+      return;
+    }
+
+    if (!AuthenticationSession.sftpFusionAuthClientSecret) {
+      logger.error(
+        'Cannot obtain new access token without sftp client secret.',
+      );
+      return;
+    }
+
+    this.fusionAuthClient.exchangeRefreshTokenForAccessToken(
+      this.refreshToken,
+      AuthenticationSession.sftpFusionAuthClientId,
+      AuthenticationSession.sftpFusionAuthClientSecret,
+      '',
+      '',
+    )
+      .then((clientResponse) => {
+        this.authToken = clientResponse.response.access_token ?? '';
```

Comment:
This `?? ''` is not safe.

The [SDK is unhelpful here](https://github.com/FusionAuth/fusionauth-typescript-client/blob/a6b1fd4844fd7242c06de812353b493c1e27979e/src/FusionAuthClient.ts#L8802-L8811), as it marks every single attribute as optional, which means that we need to ensure that the response has all the attributes we need. If the API actually did return with a "successful" response that did not include an access token, we need to consider it an error here, not later when we try to use it.

Likewise for all the other nullish coalescing operators throughout this PR.